src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Queries/CanAccessCohort/CanAccessCohortHandler.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Types;
+
+namespace SFA.DAS.CommitmentsV2.Application.Queries.CanAccessCohort
+{
+    public class CanAccessCohortHandler : IRequestHandler<CanAccessCohortQuery, bool>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+
+        public CanAccessCohortHandler(Lazy<ProviderCommitmentsDbContext> dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public Task<bool> Handle(CanAccessCohortQuery request, CancellationToken cancellationToken)
+        {
+            return _dbContext.Value.Commitment.AnyAsync(
```

Comment:
Do you think it's worth covering this query with an index?

Reply:
not required as PK is commitmentId and sql will use this